akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/DyingActor.scala

```diff
@@ -0,0 +1,32 @@
+package sample.eventstream
+
+import scala.concurrent.duration._
+import akka.actor.Actor
+import akka.actor.ActorRef
+import akka.actor.ActorLogging
+import akka.actor.Props
+import akka.actor.Cancellable
+
+object DyingActor {
+  def props(ttl: FiniteDuration): Props = Props(new DyingActor(ttl))
+  case class DyingCry(message: String)
+}
+
+class DyingActor(timeToLive: FiniteDuration) extends Actor with ActorLogging {
+  import DyingActor._
+  import context.dispatcher
+
+  // scheduled to die
+  val scheduler: Cancellable = context.system.scheduler.scheduleOnce(timeToLive)(die)
```

Comment:
in general we do not encourage that `scheduleOnce` call, it is safer to use a message to `self` and in there call the die